model_analyzer/model_manager.py

```diff
@@ -182,9 +196,17 @@ def _execute_run_configs(self):
 
         measurements = []
         while self._run_config_generator.run_configs():
+            # Check if exiting
+            if self._state_manager.exiting():
+                return measurements
+
             # Remove one run config from the list
             run_config = self._run_config_generator.next_config()
 
+            # If this run config was already run, do not run again
+            if self._already_ran_config(run_config):
+                continue
```

Comment:
I think it should not continue. It should extract the result from the state manager and append it to the list of measurements.

Reply:
Will address